src/isomorphic/React.js

```diff
@@ -86,4 +86,31 @@ var React = {
 
 };
 
+if (__DEV__) {
+  // React.DOM factories are deprecated. Set up a Proxy to listen for
+  // accesses to the React.DOM namespace and alert users to switch to
+  // the `react-addons-dom-factories` package.
+  if (typeof Proxy === 'function') {
+    var hasWarnedOfFactories = false;
```

Comment:
Maybe we should just use `Object.defineProperty` with getters that warn instead, so we don't have to special case browsers that don't yet support Proxy.